File: Algorithm/Leetcode/0000_0100/0000_0010/0010.py
# 771. Jewels and Stones

"""https://leetcode.com/problems/jewels-and-stones/discuss/113574/1-liners-PythonJavaRuby"""

class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        return sum(map(jewels.count,  stones))

File: Algorithm/Leetcode/0010.cpp
// 771. Jewels and Stones

class Solution {
private:
    static constexpr auto ansi_A = 65;
    static constexpr auto ansi_z = 122;
    size_t counter[ansi_z - ansi_A + 1];
public:
    int numJewelsInStones(string jewels, string stones) {
        for(auto &i : stones)
        {
            ++counter[i - 'A'];
        }
        size_t output = {};
                
        for(auto &i : jewels)
        {
            output += counter[i - 'A'];
        }
        return output;
    }
};